src/x509_str.c

```diff
@@ -491,12 +503,14 @@ int wolfSSL_X509_verify_cert(WOLFSSL_X509_STORE_CTX* ctx)
             ret = X509StoreAddCa(ctx->store, issuer,
                                             WOLFSSL_TEMP_CA);
             if (ret != WOLFSSL_SUCCESS) {
-                goto exit;
+                goto retry;
```

Comment:
In "retry" the depth is incremented, it appears to me that this increment is done expecting that the depth had previously been decremented on an attempt to get the chain. I think that the way this is currently setup "depth" can end up being off. For example multiple chains of depth 3 where depth is decremented a couple times then the fail to verify happens (or adding the X509 to the store fails).